packages/button/src/button-group.tsx

```diff
@@ -73,6 +73,9 @@ export const ButtonGroup = forwardRef<ButtonGroupProps, "div">((props, ref) => {
       "> *:not(:first-of-type):not(:last-of-type)": { borderRadius: 0 },
       "> *:not(:first-of-type):last-of-type": { borderStartRadius: 0 },
     }
+    if (variant === "outline") {
+      groupStyles["> *:not(:last-of-type)"] = { marginEnd: "-1px" }
+    }
```

Comment:
The issue I see with this is that we're assuming that all Chakra UI users have a variant named "outline" in their theme object.

The button component should have no knowledge of the variant named consumers to use in their design system or app. I think we should document this instead of creating a fix for it.

Reply:
Here's another way to resolve this:

- Add a `data-flushed`  attribute to the root `chakra.div` element when `isAttached` is passed to the `ButtonGroup`
- In the button theme, you can use this attribute to change the styles for the outline variant when it is within the context of that data attribute, something like.   `.chakra-button__group[data-flushed] & : {...}`